controllers/front/ProductController.php

```diff
@@ -976,6 +976,27 @@ private function getIdProductAttributeByRequest()
         return $this->tryToGetAvailableIdProductAttribute($requestedIdProductAttribute);
     }
 
+    /**
+     * Return id_product_attribute by id_product_attribute group parameter or request parameter.
+     *
+     * @return int|null
+     *
+     * @throws PrestaShopException
+     */
+    private function getIdProductAttributeByGroupOrRequest()
+    {
+        $idProductAttribute = $this->getIdProductAttributeByGroup();
+        if (null === $idProductAttribute) {
+            $idProductAttribute = (int) Tools::getValue('id_product_attribute');
+        }
+
+        if (0 === $idProductAttribute) {
+            $idProductAttribute = (int) Product::getDefaultAttribute($this->product->id);
+        }
+
+        return $idProductAttribute;
+    }
```

Comment:
Hi @sowbiba,
I think we can gain a few milliseconds, in case it is zero for example.

```php
    private function getIdProductAttributeByGroupOrRequest()
    {
        $idProductAttribute = $this->getIdProductAttributeByGroup();
        if (0 === $idProductAttribute) {
            return (int) Product::getDefaultAttribute($this->product->id);
        } elseif (null === $idProductAttribute) {
			return (int) Tools::getValue('id_product_attribute');
        } else {
            return $idProductAttribute;
        }
    }
```
What do you think ?